.github/workflows/deployment.yaml

```diff
@@ -220,25 +231,29 @@ jobs:
           do
             if [[ $FILE == *.ipynb ]]; then
               FILENAME="$(basename $FILE .ipynb)"
-              if ! az synapse notebook show --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME; then
-                cat "../../scripts/Synapse/${FILENAME}.ipynb" | envsubst > "../../scripts/Synapse/${FILENAME}Substituted.ipynb"
-                az synapse notebook create --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME --file "@../../scripts/Synapse/${FILENAME}Substituted.ipynb" --spark-pool-name sparkpool
+              if az synapse notebook show --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME; then
+                az synapse notebook delete --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME --yes
               fi
+              cat "../../scripts/Synapse/${FILENAME}.ipynb" | envsubst > "../../scripts/Synapse/${FILENAME}Substituted.ipynb"
+              az synapse notebook create --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME --file "@../../scripts/Synapse/${FILENAME}Substituted.ipynb" --spark-pool-name sparkpool
```

Comment:
I wanted to double check that this deployment has run successfully. I don't think you can delete a notebook that is already linked up to a pipeline... or at least you cannot from the console; there are some diffs from the CLI that I may not be aware of. If it has worked, feel free to ignore.

If there are changes to the pipeline notebooks, one possible solution is to re-order the way things are deleted and created: 1) delete the pipeline trigger, 2) delete the pipeline, 3) delete the notebooks, 4) create the notebooks, 5) create the pipeline, and 6) create the trigger.

Reply:
I have only run one test so far, but there were no changes to deploy so it tested a different case. This is a great point I'll investigate further.

Reply:
Just did some testing here and it appears that what @m-goggins has seen through the UI holds true with CLI. Unfortunately the `delete` command appears to succeed for notebooks attached to pipelines even though the notebook has not  truly been deleted. I'm going to modify things as your suggest.